git/odb/blob.go

```diff
@@ -0,0 +1,64 @@
+package odb
+
+import "io"
+
+// Blob represents a Git object of type "blob".
+type Blob struct {
+	// Size is the total uncompressed size of the blob's contents.
+	Size int64
+	// Contents is a reader that yields the uncompressed blob contents. It
+	// may only be read once. It may or may not implement io.ReadSeeker.
+	Contents io.Reader
+
+	// closeFn is a function that is called to free any resources held by
+	// the Blob.  In particular, this will close a file, if the Blob is
+	// being read from a file on disk.
+	closeFn func() error
+}
+
+var _ Object = (*Blob)(nil)
```

Comment:
I definitely go back and forth on whether or not they are useful. I agree that they are not useful for outside consumers, or for compile-time checks, since these types are being used in the `*ObjectDatabase` as fillers for their interface type.

That being said, I do find them very useful when looking at the code myself as a reminder of which interfaces a type is supposed to implement.